server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/formations/CapitalFormation.java

```diff
@@ -0,0 +1,179 @@
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.service.delegates.formations;
+
+import com.google.common.collect.ImmutableList;
+import lombok.*;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.lsp.cobol.service.CobolDocumentModel;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Function;
+
+public class CapitalFormation implements Formation {
+  static final String UPPERCASE = "\"Uppercase\"";
+  static final String LOWERCASE = "\"Lowercase\"";
+  private static final int MAX_LEN = 72;
+  private static final int MIN_LEN = 7;
+  private static final String PROGRAM_ID = "PROGRAM-ID";
+
+  @Data
+  private static class Context {
+    final List<TextEdit> result;
+    final Function<Character, Character> modifier;
+    int currentPos;
+    boolean reachedProgramId;
+    int dotCounter;
+    Character startedLiteralValue;
+
+    Context(Function<Character, Character> modifier) {
+      this.modifier = modifier;
+      this.result = new LinkedList<>();
+      this.reachedProgramId = false;
+      this.dotCounter = 2;
+    }
+
+    public boolean isSeekingProgramId() {
+      return (reachedProgramId && dotCounter > 0);
+    }
+
+    public void decreaseDotCounter() {
+      dotCounter--;
+    }
+
+    public void startNewLine() {
+      currentPos = MIN_LEN;
+      startedLiteralValue = null;
+    }
+
+    public boolean insideLiteralValue() {
+      return (startedLiteralValue != null);
+    }
+
+    public void checkLiteralValue(Character ch) {
+      if (ch == '"' || ch == '\'') {
+        if (startedLiteralValue == null) {
+          startedLiteralValue = ch;
+        } else {
+          if (ch.equals(startedLiteralValue)) {
+            startedLiteralValue = null;
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public List<TextEdit> format(@NonNull List<CobolDocumentModel.Line> lines, @NonNull List<String> settings) {
+    Function<Character, Character> modifier = getModifier(settings);
+    if (modifier == null) {
+      return ImmutableList.of();
+    }
+    Context context = new Context(modifier);
+
+    for (CobolDocumentModel.Line line : lines) {
+      StringBuilder builder = new StringBuilder();
+      processLine(builder, line, context);
+    }
+    return context.getResult();
+  }
+
+  private void processLine(StringBuilder builder, CobolDocumentModel.Line line, Context context) {
+    int maxLen = Math.min(line.getText().length(), MAX_LEN);
+    context.startNewLine();
+
+    // Check for commented line
+    if (maxLen >= MIN_LEN) {
+      if (line.getText().charAt(MIN_LEN - 1) == '*') {
```

Comment:
Should in-line comments be supported too?

Reply:
Good point